AArch64 build failures with static DR

#975 made static DR first-class-supported and it builds on its own and in small startstop test apps, but linking with drmemtrace it fails to link:

https://travis-ci.org/DynamoRIO/dynamorio/jobs/161201683

```
Linking CXX executable ../bin64/tool.drcacheoff.burst_static
../../lib64/debug/libdynamorio_static.a(libdynamorio_static.o): In function `check_param_bounds':
/home/travis/build/DynamoRIO/dynamorio/core/options.c:561:(.text+0x108c4): .relocation truncated to fit: R_AARCH64_LD64_GOT_LO12_NC against `main_logfile'
/home/travis/build/DynamoRIO/dynamorio/core/options.c:561:(.text+0x10934): relocation truncated to fit: R_AARCH64_LD64_GOT_LO12_NC against `main_logfile'
../../lib64/debug/libdynamorio_static.a(libdynamorio_static.o): In function `update_dynamic_options':
/home/travis/build/DynamoRIO/dynamorio/core/optionsx.h:221:(.text+0x339f8): relocation truncated to fit: R_AARCH64_LD64_GOT_LO12_NC against `main_logfile'
/home/travis/build/DynamoRIO/dynamorio/core/optionsx.h:227:(.text+0x33b28): relocation truncated to fit: R_AARCH64_LD64_GOT_LO12_NC against `main_logfile'
/home/travis/build/DynamoRIO/dynamorio/core/optionsx.h:228:(.text+0x33c58): relocation truncated to fit: R_AARCH64_LD64_GOT_LO12_NC against `main_logfile'
/home/travis/build/DynamoRIO/dynamorio/core/optionsx.h:229:(.text+0x33d88): relocation truncated to fit: R_AARCH64_LD64_GOT_LO12_NC against `main_logfile'
/home/travis/build/DynamoRIO/dynamorio/core/optionsx.h:230:(.text+0x3!3eb8): relocation truncated to fit: R_AARCH64_LD64_GOT_LO12_NC against `main_logfile'
/home/travis/build/DynamoRIO/dynamorio/core/optionsx.h:231:(.text+0x33fe8): relocation truncated to fit: R_AARCH64_LD64_GOT_LO12_NC against `main_logfile'
/home/travis/build/DynamoRIO/dynamorio/core/optionsx.h:232:(.text+0x34118): relocation truncated to fit: R_AARCH64_LD64_GOT_LO12_NC against `main_logfile'
/home/travis/build/DynamoRIO/dynamorio/core/optionsx.h:233:(.text+0x34248): relocation truncated to fit: R_AARCH64_LD64_GOT_LO12_NC against `main_logfile'
/home/travis/build/DynamoRIO/dynamorio/core/optionsx.h:234:(.text+0x34378): additional relocation overflows omitted from the output
collect2: error: ld returned 1 exit status
make[2]: *** [clients/bin64/tool.drcacheoff.burst_static] Error 1
make[1]: *** [clients/drcachesim/CMakeFiles/tool.drcacheoff.burst_static.dir/all] Error 2
```

We'll have to disable larger static tests for A64 until we understand what is going on.

This shows up in my new opcode_mix_launcher static exe as well:

https://travis-ci.org/DynamoRIO/dynamorio/jobs/418923810
```
lib64/debug/libdynamorio_static.a(libdynamorio_static.o): In function `check_param_bounds':
/work/dr/git/src/core/options.c:697:(.text+0x4350): relocation truncated to fit: R_AARCH64_LD64_GOT_LO12_NC against `main_logfile'
/work/dr/git/src/core/options.c:697:(.text+0x43c0): relocation truncated to fit: R_AARCH64_LD64_GOT_LO12_NC against `main_logfile'
lib64/debug/libdynamorio_static.a(libdynamorio_static.o): In function `update_dynamic_options':
/work/dr/git/src/core/options.c:1001:(.text+0x5284): relocation truncated to fit: R_AARCH64_LD64_GOT_LO12_NC against `main_logfile'
lib64/debug/libdynamorio_static.a(libdynamorio_static.o): In function `synchronize_dynamic_options':
/work/dr/git/src/core/options.c:2486:(.text+0x8edc): relocation truncated to fit: R_AARCH64_LD64_GOT_LO12_NC against `main_logfile'
lib64/debug/libdynamorio_static.a(libdynamorio_static.o): In function `add_shared_block':
/work/dr/git/src/core/fragment.c:391:(.text+0x1cdd8): relocation truncated to fit: R_AARCH64_LD64_GOT_LO12_NC against `main_logfile'
/work/dr/git/src/core/fragment.c:404:(.text+0x1ced0): relocation truncated to fit: R_AARCH64_LD64_GOT_LO12_NC against `main_logfile'
/work/dr/git/src/core/fragment.c:431:(.text+0x1d184): relocation truncated to fit: R_AARCH64_LD64_GOT_LO12_NC against `main_logfile'
lib64/debug/libdynamorio_static.a(libdynamorio_static.o): In function `print_shared_table_stats':
/work/dr/git/src/core/fragment.c:468:(.text+0x1d3bc): relocation truncated to fit: R_AARCH64_LD64_GOT_LO12_NC against `main_logfile'
/work/dr/git/src/core/fragment.c:469:(.text+0x1d428): relocation truncated to fit: R_AARCH64_LD64_GOT_LO12_NC against `main_logfile'
/work/dr/git/src/core/fragment.c:470:(.text+0x1d494): relocation truncated to fit: R_AARCH64_LD64_GOT_LO12_NC against `main_logfile'
/work/dr/git/src/core/fragment.c:471:(.text+0x1d500): additional relocation overflows omitted from the output
collect2: error: ld returned 1 exit status
```

--- a/clients/drcachesim/CMakeLists.txt
+++ b/clients/drcachesim/CMakeLists.txt
@@ -1,5 +1,5 @@
 # **********************************************************
-# Copyright (c) 2015-2021 Google, Inc.    All rights reserved.
+# Copyright (c) 2015-2022 Google, Inc.    All rights reserved.
 # **********************************************************
 
 # Redistribution and use in source and binary forms, with or without
@@ -235,10 +235,8 @@ set_property(GLOBAL PROPERTY DynamoRIO_drmemtrace_build_dir
 
 # We have one more example of a standalone launcher that uses raw2trace to catch
 # link errors (xref i#1409).  We just build it to test the linking; no test.
-# XXX i#2007: a binutils error causes static dynamorio to fail to link on AArch64.
-# We'll enable this once Travis and others have a fixed binutils.
 # XXX i#1997: static DR is not fully supported on Mac yet.
-if (NOT AARCH64 AND NOT APPLE)
+if (NOT APPLE)
   add_executable(opcode_mix_launcher
     tools/opcode_mix_launcher.cpp
     )
@@ -531,10 +529,9 @@ if (BUILD_TESTS)
     add_win32_flags(tool.drcacheoff.burst_aarch64_sys)
   endif ()
 
-  # FIXME i#2007: fails to link on A64
   # XXX i#1997: dynamorio_static is not supported on Mac yet
   # FIXME i#2949: gcc 7.3 fails to link certain configs
-  if (NOT AARCH64 AND NOT APPLE AND NOT DISABLE_FOR_BUG_2949)
+  if (NOT APPLE AND NOT DISABLE_FOR_BUG_2949)
     # Tests for the cache miss analyzer.
     add_executable(tool.drcachesim.miss_analyzer_unit_test tests/cache_miss_analyzer_test.cpp)
     if (ZLIB_FOUND)

--- a/suite/tests/CMakeLists.txt
+++ b/suite/tests/CMakeLists.txt
@@ -3646,11 +3646,10 @@ if (BUILD_CLIENTS)
       torunonly_drcacheoff(burst_aarch64_sys tool.drcacheoff.burst_aarch64_sys "" "@-simulator_type@basic_counts" "")
     endif ()
 
-    # FIXME i#2007: fails to link on A64
     # XXX i#1551: startstop API is NYI on ARM
     # XXX i#1997: dynamorio_static is not supported on Mac yet
     # FIXME i#2949: gcc 7.3 fails to link certain configs
-    if (NOT AARCH64 AND NOT ARM AND NOT APPLE AND NOT DISABLE_FOR_BUG_2949)
+    if (NOT ARM AND NOT APPLE AND NOT DISABLE_FOR_BUG_2949)
       set(tool.drcacheoff.burst_static_nodr ON)
       torunonly_drcacheoff(burst_static tool.drcacheoff.burst_static "" "" "")
 

--- a/suite/tests/linux/rseq.c
+++ b/suite/tests/linux/rseq.c
@@ -154,8 +154,7 @@ test_rseq_call_once(bool force_restart_in, int *completions_out, int *restarts_o
         "cmpb $0, %[force_restart]\n\t"
         "jz 7f\n\t"
         /* For -test_mode invariant_checker: expect a signal after ud2a.
-         * (An alternative is to add decoding to invariant_checker but with
-         * i#2007 and other problems that liimts the test.)
+         * (An alternative is to add decoding to invariant_checker.)
          */
         "prefetcht2 1\n\t"
         "ud2a\n\t"

